### PR TITLE
Feature/mini cart component

### PR DESCRIPTION
**Pull request by DeepaliAgrwl**
Please review the mini cart component changes and let me know your review comments

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -76,4 +76,5 @@ buildNumber.properties
 *.iml
 
 ### Mac Files ###
-*.DS_Store
\ No newline at end of file
+*.DS_Store
+*.js
```
Comment (rishabh-behl) [inline .gitignore]: @DeepaliAgrwl : why .js have been addeed to gitignore ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/minicart/clientlibs/js/minicart.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+$(document).ready(function () {
+    let cartId = '';
+    const getCountEle = document.getElementById("total-item-count");
+    getCountEle.style.display = 'none';
+    const getCookies = document.cookie;
+    if (getCookies.indexOf('cartId') > -1) {
+        const cookiesCartID = getCookies.split(';');
+        if (cookiesCartID && cookiesCartID.length > 0) {
+            cookiesCartID.forEach(function (cookiesCartIDItem) {
+                if ((cookiesCartIDItem).indexOf(cartId) > -1) {
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/minicart/clientlibs/js/minicart.js]: @AyushHCL : why do we need to iterate cookiesCartID ? # There will be only 1 cartID cookie.
@DeepaliAgrwl

File: .gitignore (hunk 1 above)
Comment (DeepaliAgrwl) [reply .gitignore]: @rishabh-behl this is committed by mistake. Reverted back.

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/minicart/minicart.html [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<a href = "${properties.cartUrl @ extension = 'html'}" id='cart-item-count'>
+<div class ='min-cart-container' onclick="redirectToCart()">
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/minicart/minicart.html]: @AyushHCL @DeepaliAgrwl : The JS code doesnt have this function

File: core/src/main/java/com/hcl/ecomm/core/config/MagentoServiceConfig.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package com.hcl.ecomm.core.config;
+
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.AttributeType;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+
+@ObjectClassDefinition(name = "Magento API's Service Configuration")
+public @interface MagentoServiceConfig {
+    String domainNameDefaultValue = "localhost/magento2/rest";
```
Comment (rishabh-behl) [inline core/src/main/java/com/hcl/ecomm/core/config/MagentoServiceConfig.java]: @DeepaliAgrwl : This should come from login config

**Review by rishabh-behl — COMMENTED** [review]
pl incorporate comments

File: core/src/main/java/com/hcl/ecomm/core/config/MagentoServiceConfig.java (hunk 4 above)
Comment (DeepaliAgrwl) [reply core/src/main/java/com/hcl/ecomm/core/config/MagentoServiceConfig.java]: change incorporated

**Pull request merged**